relationalai-sdk-snowflake/test/test_integration_snowflake.py

```diff
@@ -0,0 +1,292 @@
+
+        first_db = list_dbs_res[0]
+
+        for field in self.DATABASE_RESPONSE_FIELDS:
+            self.assertTrue(hasattr(first_db, field))
+
+        api.delete_database(self._session, db_name)
+
+    def test_get_database(self):
+        api.create_database(self._session, db_name)
+
+        get_db_res = api.get_database(self._session, db_name).collect()
+        self.assertTrue(len(get_db_res) == 1)
+
+        first_db = get_db_res[0]
+
+        for field in self.DATABASE_RESPONSE_FIELDS:
+            self.assertTrue(hasattr(first_db, field))
+
+    def test_use_database(self):
+        api.create_database(self._session, db_name)
+
+        use_database_res = api.use_database(self._session, db_name)
+        self.assertEqual(use_database_res[0][0], db_name)
+
+        current_db_res = api.get_current_database(self._session).collect()
+        self.assertEqual(current_db_res[0]["CURRENT_DATABASE"], db_name)
+
+        with self.assertRaises(Exception):
+            api.use_database(self._session, "random-db-name-that-doesnt-exist")
+
+    def tearDown(self):
+        api.delete_database(self._session, db_name)
+
+
+@unittest.skip("Skipping until we implement a new engine API that follows the async pattern.")
+class TestEngineAPI(unittest.TestCase):
+    ENGINE_RESPONSE_FIELDS = ["ACCOUNT_NAME", "CREATED_BY", "CREATED_ON", "ID", "NAME", "REGION", "SIZE", "STATE"]
+
+    def setUp(self):
+        self.session = Session.builder.configs(connection_parameters).create()
+        self.engine_name = generate_engine_name()
+
+    def test_create_engine(self):
+        print("test_create_engine", self.engine_name)
+        create_engine_res = api.create_engine(self.session, self.engine_name)
+        self.assertTrue(check_status_ok(create_engine_res))
+
+        with self.assertRaises(SnowparkSQLException):
+            api.create_engine(self.session, self.engine_name)
+
+    def test_delete_engine(self):
+        print("test_delete_engine", self.engine_name)
+        create_engine_res = api.create_engine(self.session, self.engine_name)
+        self.assertTrue(check_status_ok(create_engine_res))
+
+        delete_engine_res = api.delete_engine(self.session, self.engine_name)
+        self.assertTrue(check_status_ok(delete_engine_res))
+
+        delete_engine_res = api.delete_engine(self.session, "random-engine-name-that-doesnt-exist")
+        self.assertFalse(check_status_ok(delete_engine_res))
+
+    def test_list_engines(self):
+        print("test_list_engines", self.engine_name)
+        api.create_engine(self.session, self.engine_name)
+
+        list_engines_res = api.list_engines(self.session).collect()
+        self.assertTrue(len(list_engines_res) > 0)
+
+        first_engine = list_engines_res[0]
+
+        for field in self.ENGINE_RESPONSE_FIELDS:
+            self.assertTrue(hasattr(first_engine, field))
+
+    def test_get_engine(self):
+        api.create_engine(self.session, self.engine_name)
```

Comment:
seems to be missing the print statement here like the others

Reply:
Argh, I actually need to delete those print statements, put them there when was troubleshooting why my tests are failing with the "engine already exists" error. Turned out to be unstable create engine api. Thanks for catching this!

Reply:
the print statements have been removed!